lib/modules/platform/gitlab/index.ts

```diff
@@ -688,6 +694,24 @@ async function tryPrAutomerge(
         await setTimeout(mergeDelay * attempt ** 2); // exponential backoff
       }
 
+      if (platformPrOptions?.gitLabNotesMerge) {
+        // Use the /merge "quick action" to automerge, which will work for
+        // both merge trains or non-train merges
+        // Associated docs:
+        //  - https://docs.gitlab.com/api/notes/#create-new-merge-request-note
+        //  - https://docs.gitlab.com/user/project/quick_actions/
+        await gitlabApi.postJson(
+          `projects/${config.repository}/merge_requests/${pr}/notes`,
+          {
+            body: {
+              body: '/merge',
+              merge_request_diff_head_sha: diffHeadSha,
```

Comment:
`diffHeadSha` can be emptyif all attempts are failing. you need to handle that case

Reply:
wouldn't it just then be set to an empty string? This would result in the /notes POST call failing anyways.

Should I log an error and return false before that if diffHeadSha is empty?

Reply:
Ah it seems I misunderstood the logic then. 

The API responds 202 with an error message that the sha is missing and then just `return`s

Reply:
Would this make sense? (I'm a typescript noob, sorry)
```
      if (diffHeadSha === '') {
        throw new Error('Diff head SHA of the merge request is not set');
      }
```